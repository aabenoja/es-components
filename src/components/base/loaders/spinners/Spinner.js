import React from 'react';
import PropTypes from 'prop-types';
import styled, { keyframes } from 'styled-components';
import spinnerPath from '../../../../../public/images/spinner.svg';

const frames = keyframes`
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
`;

const Animated = styled.div`
  height: ${({ size }) => size}px;
  width: ${({ size }) => size}px;
  background-image: url(${spinnerPath});
  animation: ${frames} 1.4s linear infinite;
`;

export default function Spinner({ size = 50, ...props }) {
  return <Animated size={size} {...props} />;
}

Spinner.propTypes = {
  /**
   * The size (in pixels) of the spinner's height and width
   */
  size: PropTypes.number
};
