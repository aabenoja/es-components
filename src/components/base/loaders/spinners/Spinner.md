Any additional props are passed to the HTML element.

```
<Spinner className="some-class" />
```

```
class Sample extends React.Component {
  constructor(props) {
    super(props);

    this.state = { };
    this.onClick = this.onClick.bind(this);
  }

  onClick() {
    this.setState({ loading: true });
    setTimeout(() => this.setState({ loading: false }), 500);
  }

  render() {
    return (
      <Button handleOnClick={this.onClick}>
        { this.state.loading ? <Spinner size={24} /> : 'Click to load' }
      </Button>
    );
  }
}

<Sample />
```
