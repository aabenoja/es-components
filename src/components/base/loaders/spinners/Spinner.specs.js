/* eslint-env jest */

import React from 'react';
import renderer from 'react-test-renderer';

jest.mock('../../../../../public/images/spinner.svg', () => ({}));

import Spinner from './Spinner';

describe('spinner component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Spinner className="foo bar" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
