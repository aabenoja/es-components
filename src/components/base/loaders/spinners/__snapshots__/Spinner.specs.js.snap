exports[`spinner component renders correctly 1`] = `
<div
  className="foo bar iPWAxi"
  size={50} />
`;
